frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -898,6 +900,11 @@ pub mod pallet {
             Self::ensure_pallet_enabled()?;
             let staker = ensure_signed(origin)?;
 
+            ensure!(
+                Ledger::<T>::contains_key(&staker),
+                Error::<T>::NotActiveStaker
+            );
```

Comment:
We should keep it to 1 read and 1 write - `contains_key` counts as a write now.

I'd suggest to read the value from the DB, check it using the `is_empty()` method and then insert the modified value.